packages/sandbox/src/file_watching_sandbox.ts

```diff
@@ -184,16 +184,19 @@ export class FileWatchingSandbox extends EventEmitter implements Sandbox {
         this.printer.log(
           "[Sandbox] Detected file changes while previous deployment was in progress. Invoking 'sandbox' again"
         );
-        await this.deploy(options);
+        deployResult = await this.deploy(options);
       }
       latch = 'open';
 
-      // Idle state, let customers know and start streaming function logs
+      // Idle state, let customers know
       this.emitWatching();
-      await this.functionsLogStreamer.startStreamingLogs(
-        await this.backendIdSandboxResolver(options.identifier),
-        options.functionStreamingOptions
-      );
+      // if deployment was successful, start streaming function logs
+      if (deployResult) {
+        await this.functionsLogStreamer.startStreamingLogs(
+          await this.backendIdSandboxResolver(options.identifier),
+          options.functionStreamingOptions
+        );
+      }
```

Comment:
This will prevent the log streaming to work after every failed deployment which we might not want to do. After a successful deployment, it should be always ok to let the log streamer working as many of the failures are simply syntax errors where customers are making live updates to their app.